build-info-extractor-docker/src/main/java/org/jfrog/build/extractor/docker/types/DockerImage.java

```diff
@@ -257,41 +259,20 @@ private DockerLayers getLayers(ArtifactoryDependenciesClient dependenciesClient,
      */
     private void findAndSetManifestFromArtifactory(ArtifactoryDependenciesClient dependenciesClient, Log logger, DockerUtils.CommandType cmdType) throws IOException {
         // Try to get manifest, assuming reverse proxy
-        String proxyImagePath = DockerUtils.getImagePath(imageTag);
-        try {
-            logger.info("Trying to fetch manifest from Artifactory, assuming reverse proxy configuration.");
-            checkAndSetManifestAndImagePathCandidates(proxyImagePath, proxyImagePath, dependenciesClient);
-            return;
-        } catch (IOException e) {
-            logger.error("The manifest could not be fetched from Artifactory, assuming reverse proxy configuration - " + e.getMessage());
-            // Ignore - Artifactory may have a proxy-less setup. Let's try that.
-        }
-        // Try to get manifest, assuming proxy-less
-        String proxyLessImagePath = proxyImagePath.substring(proxyImagePath.indexOf("/") + 1);
-        logger.info("Trying to fetch manifest from Artifactory, assuming proxy-less.");
-        try {
-            checkAndSetManifestAndImagePathCandidates(proxyLessImagePath, proxyLessImagePath, dependenciesClient);
-        } catch (IOException e) {
-            logger.error("The manifest could not be fetched from Artifactory - " + e.getMessage());
-            // If image path includes more than 3 slashes, Artifactory doesn't store this image under 'library',
-            // thus we should not look further.
-            int totalSlash = StringUtils.countMatches(proxyImagePath, "/");
-            if (cmdType == DockerUtils.CommandType.Push || totalSlash > 3) {
-                throw e;
+        String ImagePath = DockerUtils.getImagePath(imageTag);
+        ArrayList<String> manifestPathCandidate = new ArrayList<>(DockerUtils.getArtManifestPath(ImagePath, targetRepo, cmdType));
+        logger.info("Trying to fetch manifest from Artifactory.");
+        int listLen = manifestPathCandidate.toArray().length;
+        for (int i = 0; i < listLen; i++) {
+            try {
+                checkAndSetManifestAndImagePathCandidates(manifestPathCandidate.get(i), dependenciesClient);
+                return;
+            } catch (IOException e) {
+                // Ignore - unless we have no search path.
```

Comment:
I suggest changing this comment to the following, but please see if this is what you actually wanted to say here:
// Throw the exception only if we reached the end of the loop, which means we tried all options.